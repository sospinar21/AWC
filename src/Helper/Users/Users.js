import { Config, CognitoUserPool, CognitoUserAttribute, CognitoUser, AuthenticationDetails, CognitoIdentityCredentials } from 'amazon-cognito-identity-js';
import { Lambda } from 'aws-sdk';

var poolData = { UserPoolId : 'us-west-2_t4LvOKjcE',
  ClientId: '47bsfajnf2rmvpt02q8qjvm29u'
};
var userPool = new CognitoUserPool(poolData);

export const checkUser = () => {
  var cognitoUser = userPool.getCurrentUser();
  return cognitoUser;
}

export const signUp = (user) => {
  var attributeList = [];

  var dataEmail = {
    Name : 'email',
    Value : user.email
  };
  var city = {
    Name : 'custom:city',
    Value : user.city
  };
  var attributeEmail = new CognitoUserAttribute(dataEmail);
  var attributeCity = new CognitoUserAttribute(city);

  attributeList.push(attributeEmail);
  attributeList.push(attributeCity);

  userPool.signUp(user.email, user.password, attributeList, null, function(err, result) {
    if (err) {
      const error = document.querySelector('.created');
      error.innerText = 'invalid email or password: min 6 characters';
      return;
    }
    var cognitoUser = result.user;
    const created = document.querySelector('.created');
    created.innerText = 'Welcome! your new user name is ' + cognitoUser.getUsername();
    window.location.href = "http://awc.dance/";
  });
};

export const logIn = (user) => {
  var userData = {Username: user.email, Pool: userPool};
  var cognitoUser = new CognitoUser(userData);

  var authenticationData = {Username: user.email, Password: user.password};
  var authenticationDetails = new AuthenticationDetails(authenticationData);

  cognitoUser.authenticateUser(authenticationDetails, {
    onSuccess: function (result) {
      const success = document.querySelector('.success');
      success.innerText = 'Welcome back! ' + cognitoUser.getUsername();
      window.location.href = "http://awc.dance/";
      return result; 
    },

    onFailure: function (err) {
      const incorrect = document.querySelector('.success');
      incorrect.innerText = 'Check your Password or Email';
      return err; 
    }

  });
};

export const forgotPassword = () => {
  CognitoUser.forgotPassword({
    onSuccess: function (result) {
      console.log('call result: ' + result);
    },
    onFailure: function(err) {
      alert(err);
    },
    inputVerificationCode() {
      var verificationCode = prompt('Please input verification code ', '');
      var newPassword = prompt('Enter new password ', '');
      CognitoUser.confirmPassword(verificationCode, newPassword, this);
    }
  });
};

export const userSignout = () => {
  var cognitoUser = userPool.getCurrentUser();
  console.log(cognitoUser)
  cognitoUser.signOut();
  console.log(cognitoUser)
  return cognitoUser;
}

