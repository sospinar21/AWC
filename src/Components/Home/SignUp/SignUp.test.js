import {SignUp, mapStateToProps, mapDispatchToProps} from '../signUp/signUp';
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import * as users from '../../../Helper/Users/Users';

describe('signUp', () => {

  let signUp;
  let mockProps;
  let event;

  beforeEach(() => {

    mockProps = {
      suggestedEvents: [{name: 'party'}],
    };

    signUp = shallow(<SignUp {...mockProps}/>, { disableLifecycleMethods: true });
    users.signUp = jest.fn();
  });

  it('matches snapshot', () => {
    expect(signUp).toMatchSnapshot();
  });


  describe('handleInputChange', () => {

    it('sets the state with the new users email', () => {
      
      event = {
        target: {
          name: 'email',
          value: 'user@user.com'
        },
        preventDefault: () => {}
      };

      signUp.instance().handleInputChange(event);

      expect(signUp.state().email).toEqual('user@user.com');
    });

    it('sets the state with the new users password', () => {
      
      event = {
        target: {
          name: 'password',
          value: 'hola!'
        },
        preventDefault: () => {}
      };

      signUp.instance().handleInputChange(event);

      expect(signUp.state().password).toEqual('hola!');
    });
  });

  describe('userSignUp', () => {

    it('clears the state', () => {
      
      event = {
        preventDefault: () => {}
      };

      signUp.instance().userSignUp(event);

      expect(signUp.state().password).toEqual('');
      expect(signUp.state().email).toEqual('');
      
    });
  });

  describe('validateEmail', () => {

    it('returns true if the length of email or password is 0', () => {

      signUp.state().email = '';
      signUp.state().password = '';

      expect(signUp.instance().validateEmail()).toEqual(true);
    });
  });

  describe('mapStateToProps', () => {

    let mappedProps

    it('returns an object with the user', () => {

      const mockState = {
        suggestedEvents: [{name: "workshop"}]
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
      
    })
  })
});