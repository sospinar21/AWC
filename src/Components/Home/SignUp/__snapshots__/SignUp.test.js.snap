// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signUp matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    suggestedEvents={
        Array [
            Object {
              "name": "party",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          SIGN UP
</h3>,
        <div
          className="logo"
>
          <div
                    className="logocir"
          >
                    AWC
          </div>
</div>,
        <h4
          className="created"
>
           
</h4>,
        <form
          autoComplete="on"
          className="form"
>
          <input
                    autoComplete="on"
                    className="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
                    value=""
          />
          <input
                    autoComplete="on"
                    className="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <input
                    autoComplete="on"
                    className="city"
                    name="city"
                    onChange={[Function]}
                    placeholder="City"
                    value=""
          />
          <button
                    className="signup"
                    disabled={true}
                    onClick={[Function]}
          >
                    Sign Up 
          </button>
</form>,
        <div
          className="policy"
>
          <p>
                    By signing up, you agree to our Terms, Data Policy and Cookies Policy.
          </p>
</div>,
      ],
      "className": "signup-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SIGN UP",
        },
        "ref": null,
        "rendered": "SIGN UP",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="logocir"
>
            AWC
</div>,
          "className": "logo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "AWC",
            "className": "logocir",
          },
          "ref": null,
          "rendered": "AWC",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ",
          "className": "created",
        },
        "ref": null,
        "rendered": " ",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "on",
          "children": Array [
            <input
              autoComplete="on"
              className="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
/>,
            <input
              autoComplete="on"
              className="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            <input
              autoComplete="on"
              className="city"
              name="city"
              onChange={[Function]}
              placeholder="City"
              value=""
/>,
            <button
              className="signup"
              disabled={true}
              onClick={[Function]}
>
              Sign Up 
</button>,
          ],
          "className": "form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "on",
              "className": "email",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "on",
              "className": "password",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "on",
              "className": "city",
              "name": "city",
              "onChange": [Function],
              "placeholder": "City",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up ",
              "className": "signup",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign Up ",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            By signing up, you agree to our Terms, Data Policy and Cookies Policy.
</p>,
          "className": "policy",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "By signing up, you agree to our Terms, Data Policy and Cookies Policy.",
          },
          "ref": null,
          "rendered": "By signing up, you agree to our Terms, Data Policy and Cookies Policy.",
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            SIGN UP
</h3>,
          <div
            className="logo"
>
            <div
                        className="logocir"
            >
                        AWC
            </div>
</div>,
          <h4
            className="created"
>
             
</h4>,
          <form
            autoComplete="on"
            className="form"
>
            <input
                        autoComplete="on"
                        className="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="email"
                        value=""
            />
            <input
                        autoComplete="on"
                        className="password"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <input
                        autoComplete="on"
                        className="city"
                        name="city"
                        onChange={[Function]}
                        placeholder="City"
                        value=""
            />
            <button
                        className="signup"
                        disabled={true}
                        onClick={[Function]}
            >
                        Sign Up 
            </button>
</form>,
          <div
            className="policy"
>
            <p>
                        By signing up, you agree to our Terms, Data Policy and Cookies Policy.
            </p>
</div>,
        ],
        "className": "signup-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SIGN UP",
          },
          "ref": null,
          "rendered": "SIGN UP",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="logocir"
>
              AWC
</div>,
            "className": "logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "AWC",
              "className": "logocir",
            },
            "ref": null,
            "rendered": "AWC",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
            "className": "created",
          },
          "ref": null,
          "rendered": " ",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "on",
            "children": Array [
              <input
                autoComplete="on"
                className="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value=""
/>,
              <input
                autoComplete="on"
                className="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <input
                autoComplete="on"
                className="city"
                name="city"
                onChange={[Function]}
                placeholder="City"
                value=""
/>,
              <button
                className="signup"
                disabled={true}
                onClick={[Function]}
>
                Sign Up 
</button>,
            ],
            "className": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "on",
                "className": "email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "on",
                "className": "password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "on",
                "className": "city",
                "name": "city",
                "onChange": [Function],
                "placeholder": "City",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up ",
                "className": "signup",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Up ",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              By signing up, you agree to our Terms, Data Policy and Cookies Policy.
</p>,
            "className": "policy",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "By signing up, you agree to our Terms, Data Policy and Cookies Policy.",
            },
            "ref": null,
            "rendered": "By signing up, you agree to our Terms, Data Policy and Cookies Policy.",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
